### PR TITLE
Change IIS projects to target netcoreapp3.0

### PR DESCRIPTION
**Pull request by Tratcher**
#3754
Also fixed an error in HttpResponseStream.cs. The NETCOREAPP2_1 targets weren't getting compiled since the assembly only targeted netstandard2.0.

### CODE REVIEW

**Comment by jkotalik** [comment]
I think the test projects may be broken for publish. PublishedSitesFixture.cs publishes for netcoreapp2.2 still in tests.

Note: tests are still disabled in master due to a websdk issue causing all out of proc tests to fail.

**Comment by pakrym** [comment]
We need to re-enable tests before taking this.

**Comment by Tratcher** [comment]
What will it take to re-enable tests? This is blocking lower repos from getting converted.

**Comment by jkotalik** [comment]
I think @pakrym is looking into it now.

**Review by natemcmaster — APPROVED** [review]

**Review by halter73 — APPROVED** [review]

**Comment by jkotalik** [comment]
Just got new hosting dependencies, will update shortly.

**Comment by natemcmaster** [comment]
We're adding required PR checks to ensure the repo compiles before you merge. Can you update this PR by rebasing on master?

```
git fetch origin master:master
git rebase master
```

Alternative: if you plan to 'squash merge' this PR, you can also use a merge to update this PR. This can be easier to manage if you have lots of changes in this PR.
```
git fetch origin master:master
git rebase master
```

**Comment by natemcmaster** [comment]
By the way, you can ignore the error that says "AspNetCore-pr-validation — Unable to find '.azure/pipelines/fast-pr-validation.yml'". This is the result of changing PR checks after a PR was created. It shouldn't appear in PRs created from now on.

File: src/IISIntegration/build/testsite.props [hunk 1]
```diff
@@ -44,13 +44,6 @@
     <Copy SourceFiles="$(MSBuildThisFileDirectory)launchSettings.json" DestinationFolder="$(MSBuildProjectDirectory)\Properties" SkipUnchangedFiles="true" />
   </Target>
 
-  <!-- Deps file injection-->
-  <!--<ItemGroup Condition="('$(InProcessTestSite)' == 'true') AND ('$(ReferenceTestTasks)' != 'false')">
```
Comment (pakrym) [inline src/IISIntegration/build/testsite.props]: Keep this with a comment.

File: src/IISIntegration/test/Common.FunctionalTests/CompressionTests.cs [hunk 2]
```diff
@@ -49,8 +49,13 @@ public async Task PassesThroughCompression()
             Assert.Equal("gzip", response.Content.Headers.ContentEncoding.Single());
             Assert.Equal(
                 new byte[] {
-                    0x1F, 0x8B, 0x08, 0x00, 0x00, 0x00, 0x00, 0x00,
-                    0x04, 0x0B, 0x63, 0x60, 0xA0, 0x3D, 0x00, 0x00,
+                    0x1F, 0x8B,
```
Comment (pakrym) [inline src/IISIntegration/test/Common.FunctionalTests/CompressionTests.cs]: Just replace the byte.

**Comment by jkotalik** [comment]
Tests should be passing now.

**Comment by Tratcher** [comment]
Still quite a few test failures where it looks like the server is failing to start.

**Comment by jkotalik** [comment]
https://github.com/aspnet/AspNetCore/issues/4512 for flaky test.

File: src/IISIntegration/test/Common.FunctionalTests/Inprocess/ClientDisconnectTests.cs [hunk 3]
```diff
@@ -66,7 +66,7 @@ public async Task RequestAbortedTokenFires()
             await _fixture.Client.RetryRequestAsync("/WaitingRequestCount", async message => await message.Content.ReadAsStringAsync() == "0");
         }
 
-        [ConditionalFact]
+        [ConditionalFact(Skip = "https://github.com/aspnet/AspNetCore/issues/4512")]
```
Comment (pakrym) [inline src/IISIntegration/test/Common.FunctionalTests/Inprocess/ClientDisconnectTests.cs]: We should undo this.

**Pull request merged**